MarsRover/Validator.cs

```diff
@@ -6,33 +6,37 @@ namespace MarsRover
 {
     public static class Validator
     {
-        public static bool IsValidMap(string input) // TODO: refactor
+        public static bool IsValidMap(string input)
         {
             var mapElements = SplitInput(input, "\n");
-            // if has less than 2 elements, return false
-            // count the length of each line, if all !=, return false
-            // loop through each element 
-            // split into char array
-            // if doesn't contain only 'N' or 'O' return false
-            // return true
             var numberOfElements = mapElements.Length;
             if (numberOfElements == 1) return false;
             var elementList = new List<int>();
             var mapCharsCount = 0;
             var count = 0;
+            CountValidChars(mapElements, elementList, ref mapCharsCount, ref count);

```

Comment:
Is there any reason why you chose to use references instead of values? i.e. why not return a count and store it in a variable